### ISSUE TITLE
[Bug]: NC24.0.1 Files and WebDAV doesn't work after update

### ISSUE THREAD

**Issue by Termindiego25** [nextcloud/server#32609]
### ⚠️ This issue respects the following points: ⚠️

- [X] This is a **bug**, not a question or a configuration/webserver/proxy issue.
- [X] This issue is **not** already reported on Github _(I've searched it)_.
- [X] Nextcloud Server **is** up to date. See [Maintenance and Release Schedule](https://github.com/nextcloud/server/wiki/Maintenance-and-Release-Schedule) for supported versions.
- [X] I agree to follow Nextcloud's [Code of Conduct](https://nextcloud.com/contribute/code-of-conduct/).

### Bug description

On monday, my NC showed an available update at the server settings and I tried to install it. First of all, I updated from 23.0.4 to 23.0.5 without errors, and then showed another update to 24.0.1 version, installed withour errors but with some incompatible apps.

Once all was updated, I checked the security controls of the dashboard and it showed that opcache is needed and a “x-download-options” header is missing. I tried to add it without any shown errors and the checks work fine.

When I tried to access to the website with the main domain, it doesn’t works and show a message of an internal error and inviting to check the logs. Then I tried to access with an administrator account and the error is out, but when I tried to access my files, they are not working. Using the website, I can list the folders at the first level of my user files folder, but when I click one the page refreshes without showing the content and showing again the same. Also, trying with WebDAV, Windows or Android client doesn’t work too because they can’t connect to my NC server.

I also tried to remove the “x-download-options” header, repair tree, repair installation, scan folders… with the OCC commands but I could not repair the problem.

Can you help me to solve this problem? Any information you need, do not hesitate to ask for them.
Thank you

### Steps to reproduce

(what I have did)
1. Update from 23.0.4 to 23.0.5, and then to 24.0.1
2. Add the “x-download-options” to the apache configuration like the Nginx recommended at the config guide (with the noopen)
3. Add php-opcache package to the system

### Expected behavior

Once all is updated and installed, I should be able to access all files and folders in my server using website, and WebDAV, Android or Windows app should be able to connect to the server instead of throw a connection error.
Also, I should be able to login with no administrator users instead of receive an internal error and an invitation to contact an administrator to check logs

### Installation method

Web installer on a VPS or web space

### Operating system

Debian/Ubuntu

### PHP engine version

PHP 8.0

### Web server

Apache (supported)

### Database engine version

MariaDB

### Is this bug present after an update or on a fresh install?

Updated to a major version (ex. 22.2.3 to 23.0.1)

### Are you using the Nextcloud Server Encryption module?

Encryption is Disabled

### What user-backends are you using?

- [X] Default user-backend _(database)_
- [ ] LDAP/ Active Directory
- [ ] SSO - SAML
- [ ] Other

### Configuration report

```shell
{
    "system": {
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "termincloud.diegosr.es",
            "192.168.1.118",
            "terminx.diegosr.es",
            "mre.diegosr.es"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "24.0.1.1",
        "overwrite.cli.url": "https:\/\/termincloud.diegosr.es",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "default_phone_region": "ES",
        "memcache.local": "\\OC\\Memcache\\Memcached",
        "memcache.distributed": "\\OC\\Memcache\\Memcached",
        "memcached_servers": [
            [
                "localhost",
                11211
            ]
        ],
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpmode": "smtp",
        "mail_sendmailmode": "smtp",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpauth": 1,
        "mail_smtpauthtype": "LOGIN",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "app_install_overwrite": [
            "ocr",
            "pdfdraw",
            "customproperties",
            "files_external_onedrive",
            "camerarawpreviews",
            "documentserver_community",
            "files_3d",
            "imageconverter",
            "hsts"
        ],
        "maintenance": false,
        "theme": "",
        "loglevel": "1",
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    }
}
```

### List of activated Apps

```shell
Enabled:
  - accessibility: 1.10.0
  - activity: 2.16.0
  - admin_audit: 1.14.0
  - bruteforcesettings: 2.4.0
  - calendar: 3.3.1
  - camerarawpreviews: 0.7.15
  - cfg_share_links: 2.0.0
  - circles: 24.0.0
  - cloud_federation_api: 1.7.0
  - comments: 1.14.0
  - contacts: 4.1.1
  - contactsinteraction: 1.5.0
  - dashboard: 7.4.0
  - dav: 1.22.0
  - deck: 1.7.0
  - documentserver_community: 0.1.12
  - extract: 1.3.5
  - federatedfilesharing: 1.14.0
  - federation: 1.14.0
  - files: 1.19.0
  - files_automatedtagging: 1.14.0
  - files_downloadactivity: 1.13.0
  - files_external: 1.16.1
  - files_markdown: 2.3.6
  - files_mindmap: 0.0.26
  - files_pdfviewer: 2.5.0
  - files_rightclick: 1.3.0
  - files_sharing: 1.16.2
  - files_trashbin: 1.14.0
  - files_versions: 1.17.0
  - files_videoplayer: 1.13.0
  - files_zip: 1.1.1
  - firstrunwizard: 2.13.0
  - forms: 2.5.0
  - hsts: 0.9.0
  - imageconverter: 1.3.2
  - integration_google: 1.0.6
  - integration_onedrive: 1.1.2
  - logreader: 2.9.0
  - lookup_server_connector: 1.12.0
  - mail: 1.12.0
  - maps: 0.1.10
  - metadata: 0.16.0
  - ncdownloader: 0.9.33
  - nextcloud_announcements: 1.13.0
  - notes: 4.3.1
  - notifications: 2.12.0
  - oauth2: 1.12.0
  - onlyoffice: 7.3.4
  - password_policy: 1.14.0
  - photos: 1.6.0
  - previewgenerator: 5.0.0
  - privacy: 1.8.0
  - provisioning_api: 1.14.0
  - ransomware_protection: 1.13.0
  - recommendations: 1.3.0
  - serverinfo: 1.14.0
  - settings: 1.6.0
  - sharebymail: 1.14.0
  - side_menu: 2.3.5
  - spreed: 14.0.1
  - support: 1.7.0
  - survey_client: 1.12.0
  - systemtags: 1.14.0
  - tasks: 0.14.4
  - text: 3.5.1
  - theming: 1.15.0
  - theming_customcss: 1.11.0
  - timetracker: 0.0.74
  - transfer: 0.5.0
  - twofactor_backupcodes: 1.13.0
  - updatenotification: 1.14.0
  - user_status: 1.4.0
  - video_converter: 1.0.5
  - viewer: 1.8.0
  - weather_status: 1.4.0
  - workflow_ocr: 1.24.1
  - workflowengine: 2.6.0
Disabled:
  - customproperties: 2.0.4
  - drawio: 1.0.2
  - encryption
  - epubreader: 1.4.7
  - facerecognition
  - files_3d: 0.5.0
  - files_antivirus: 3.2.2
  - music: 1.5.2
  - occweb: 0.1.0
  - ocdownloader: 1.8.0
  - ocr: 6.0.58
  - scanner: 0.2.2
  - signlive: 0.1.11
  - user_ldap
```

### Nextcloud Signing status

```shell
No errors have been found.
```

### Nextcloud Logs

```shell
Link to nextcloud downloaded log: https://www.diegosr.es/nextcloud.log
```

### Additional info

[Link to Apache access log](https://pastebin.com/Mu1UmPbw)
[Link to Apache error log](https://pastebin.com/A6PTXy5L)

**Comment by effatee** [comment]
Exact same upgrade context (from 23.0.4 to 23.0.5 to to 24.0.1)
Getting a "loading" circle forever when trying to display files from the UI. Waiting will do nothing.
After some insistance (reload, click on the "house" breadcrumb), files are eventually displayed.
WebDav works fine.

**Comment by KunaPrime** [comment]
can confirm the same error I can't go into the instance and the only error that I get is 

```
Internal Server Error

The server was unable to complete your request.

If this happens again, please send the technical details below to the server administrator.

More details can be found in the server log.
```

**Comment by Ornanovitch** [comment]
> Using the website, I can list the folders at the first level of my user files folder, but when I click one the page refreshes without showing the content and showing again the same

>  Getting a "loading" circle forever when trying to display files from the UI. Waiting will do nothing.
After some insistance (reload, click on the "house" breadcrumb), files are eventually displayed.
WebDav works fine.

@Termindiego25 @effatee About this particular behavior, do you have the same error than I got into the web console? See here:
- #32582

For what's left I have no problem with WebDAV neither internal server error.

**Comment by Termindiego25** [comment]
> > Using the website, I can list the folders at the first level of my user files folder, but when I click one the page refreshes without showing the content and showing again the same
> 
> > Getting a "loading" circle forever when trying to display files from the UI. Waiting will do nothing.
> > After some insistance (reload, click on the "house" breadcrumb), files are eventually displayed.
> > WebDav works fine.
> 
> @Termindiego25 @effatee About this particular behavior, do you have the same error than I got into the web console? See here:
> 
> * [[Bug]: files index load indefinitely -> `Uncaught (in promise) TypeError: e.vm is null` #32582](https://github.com/nextcloud/server/issues/32582)
> 
> For what's left I have no problem with WebDAV neither internal server error.

This is the console log of Chrome when I try to access to the website as a user (and shows the internal error):
[Link to the log](https://pastebin.com/bGkbBFjF)

And this is the console log trying to acces a folder in website Files with an administrator user (because a normal user shows the internal error):
[Link to the log](https://pastebin.com/kgiJiubc)

**Comment by Ornanovitch** [comment]
> This is the console log of Chrome when I try to access to the website as a user (and shows the internal error)

OK it doesn't seems to be related to my own issue. But regarding the log you pasted, maybe try disabling the dashboard app and see if it works better.

> And this is the console log trying to acces a folder in website Files with an administrator user (because a normal user shows the internal error)

The only clear error I see is a 404 on a file. Maybe you could try an `occ files:scan` on your server.

**Comment by Termindiego25** [comment]
> > This is the console log of Chrome when I try to access to the website as a user (and shows the internal error)
> 
> OK it doesn't seems to be related to my own issue. But regarding the log you pasted, maybe try disabling the dashboard app and see if it works better.
> 
> > And this is the console log trying to acces a folder in website Files with an administrator user (because a normal user shows the internal error)
> 
> The only clear error I see is a 404 on a file. Maybe you could try an `occ files:scan` on your server.

I tried it, including repair-tree but it didn't solve the problem

**Comment by effatee** [comment]
> > Using the website, I can list the folders at the first level of my user files folder, but when I click one the page refreshes without showing the content and showing again the same
> 
> > Getting a "loading" circle forever when trying to display files from the UI. Waiting will do nothing.
> > After some insistance (reload, click on the "house" breadcrumb), files are eventually displayed.
> > WebDav works fine.
> 
> @Termindiego25 @effatee About this particular behavior, do you have the same error than I got into the web console? See here:
> 
> * [[Bug]: files index load indefinitely -> `Uncaught (in promise) TypeError: e.vm is null` #32582](https://github.com/nextcloud/server/issues/32582)
> 
> For what's left I have no problem with WebDAV neither internal server error.

I have same issues indeed, but no error is raised.
Also appeared since upgrade to 24.0.1

**Comment by Termindiego25** [comment]
Any ideas to try to solve the problem? I can't use my server for two weeks

**Comment by KunaPrime** [comment]
at my instance I can access `<nexcloud-instance>/apps/calendar`, `<nexcloud-instance>/apps/mail` but not `<nexcloud-instance>/apps/dashboard` or my file at all

**Comment by maxmenGMK** [comment]
I have the same error but this error is only for one account. I can to log in to NC with one account but cant with another one. Just yesterday all was fine.
I disabled dashboard app but the problem still exist.

    ID Запроса: hyuiVsRU4Z1lz3wNSRju
    Тип: Exception
    Код: 0
    Сообщение: User folder for user1@mydomain.ru exists as a file
    Файл: «/var/www/html/nextcloud/lib/private/Files/Node/Root.php»
    Строка: 389


Трассировка

#0 [internal function]: OC\Files\Node\Root->getUserFolder()
#1 /var/www/html/nextcloud/lib/private/Files/Node/LazyFolder.php(72): call_user_func_array()
#2 /var/www/html/nextcloud/lib/private/Files/Node/LazyRoot.php(40): OC\Files\Node\LazyFolder->__call()
#3 /var/www/html/nextcloud/lib/private/Server.php(1658): OC\Files\Node\LazyRoot->getUserFolder()
#4 /var/www/html/nextcloud/apps/files/lib/Activity/Helper.php(60): OC\Server->getUserFolder()
#5 /var/www/html/nextcloud/apps/files/lib/Controller/ViewController.php(210): OCA\Files\Activity\Helper->getFavoriteFilePaths()
#6 /var/www/html/nextcloud/lib/private/AppFramework/Http/Dispatcher.php(225): OCA\Files\Controller\ViewController->index()
#7 /var/www/html/nextcloud/lib/private/AppFramework/Http/Dispatcher.php(133): OC\AppFramework\Http\Dispatcher->executeController()
#8 /var/www/html/nextcloud/lib/private/AppFramework/App.php(172): OC\AppFramework\Http\Dispatcher->dispatch()
#9 /var/www/html/nextcloud/lib/private/Route/Router.php(298): OC\AppFramework\App::main()
#10 /var/www/html/nextcloud/lib/base.php(1023): OC\Route\Router->match()
#11 /var/www/html/nextcloud/index.php(36): OC::handleRequest()
#12 {main}

**Comment by Termindiego25** [comment]
Good morning, I have tried to update to 24.0.2 and I have the same troubles.
The update was installed without errors

**Comment by szijl** [comment]
i found that users can login after disabling the File sharing app using the admin account and the sync starts again using the Nextcloup Desktop App's. The files app doesn't show files: folder not available. Updating to 24.0.2. did not fix these issues.

**Comment by szijl** [comment]
> > i found that users can login after disabling the File sharing app using the admin account and the sync starts again using the Nextcloup Desktop App's. The files app doesn't show files: folder not available. Updating to 24.0.2. did not fix these issues.
> 
> I have tried disabling Files sharing app, Desktop & Android apps are working again, and also not admin users can log in again. As you said, on web interface Files app show an error saying "Folder not available" and WebDAV also doesn't work. Thank you for the suggestion, now I can use some parts of my server again.
> 
> Could be an incompatibility of that app with the new version or something wrong installed?

After applying https://github.com/nextcloud/server/pull/32924 resolved the issue "Folder not available" for me.

**Comment by Termindiego25** [comment]
> > > i found that users can login after disabling the File sharing app using the admin account and the sync starts again using the Nextcloup Desktop App's. The files app doesn't show files: folder not available. Updating to 24.0.2. did not fix these issues.
> > 
> > 
> > I have tried disabling Files sharing app, Desktop & Android apps are working again, and also not admin users can log in again. As you said, on web interface Files app show an error saying "Folder not available" and WebDAV also doesn't work. Thank you for the suggestion, now I can use some parts of my server again.
> > Could be an incompatibility of that app with the new version or something wrong installed?
> 
> After applying #32924 resolved the issue "Folder not available" for me.

I have tried the following steps extracted by the issue mentioned:
 - Go to /{nextcloud-intallation}/lib/private/Memcache/Memcached.php
 - Find line "//\Memcached::OPT_BINARY_PROTOCOL =>      true,"
 - Uncomment it leaving as "\Memcached::OPT_BINARY_PROTOCOL =>      true,"
 - Save and restart

With that steps, WebDAV and website are working again, showing the folders and content as expected.
Maybe this apps need some code review in order to solve it in the next update

**Comment by claneys** [comment]
Thanks @Termindiego25 , uncommenting the line solved the problem for me.

**Comment by claneys** [comment]
How this issue could be fixed correctly ?

I'm trying to move to container and as the container is quite ephemeral, this fix should be applied every time manually. It is not viable way to solve this issue. Could it be fixed upstream instead .

**Comment by claneys** [comment]
v24.0.3 is fine and got the fix. Thanks :)

**Comment by Termindiego25** [comment]
I have tried to revert the changes previously commented and I have updated server to v24.0.3 in order to check if this was fixed.
Now is working fine, so you don't need to do that, just update to solve it

**Comment by tdaly001** [comment]
Running nextcloud pi v1.52.0 nextcloud version 25.0.5.1 I was getting the error about web server not configured and WebDAV interface seems to be broken.  This occurred without any obvious cause.  I shut the server down properly due to a tornado nearby, when I brought it back up, the sync clients did not connect and the android app said the server was in maintenance mode.  The web interface had the 503 error, the control panel still worked.

I tried and reversed so many things that I am not sure how I got here, but in the /etc/apache2/sites-available/nextcloud.conf file there was a line with "DAV off".  I changed it to "DAV on" and rebooted.  I needed to use the clear-php-opcache tool in the control panel, but then all returned.  I am very worried this will recur, so any ideas as to why this happened would be very welcome.  If this helps anyone else, I have found treasure troves of answers in these (github in general) forums and hope I can contribute.

Furthermore, I had previously had the error in the following post. 
https://github.com/nextcloud/nextcloudpi/issues/1593
I commented the lines as directed and it worked again for about 2 weeks.  This behavior was similar, so I began here, but the lines I had commented out of /var/www/nextcloud/config/config.php where mysteriously gone.  Keep in mind, no updates had been run.
Thank you all and hopefully this can help some else.
